go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/featureform
 
-go 1.21
+go 1.22
+
+toolchain go1.23.3
```

Comment:
Any reason we're setting a newer toolchain? And if so, any reason not to just bump the version to 1.23?

Reply:
Was modified when pulling the arrow go library v17. But...I'm going to try and revert to `1.21` and see if it still works, since their repo's [go.mod](https://github.com/apache/arrow/blob/go/v17.0.0/go/go.mod) apparently only needs 1.21.

Reply:
cleaned up the mod file and 1.21 works just fine.